Dockerfile

```diff
@@ -126,6 +126,9 @@ RUN --mount=type=bind,from=build,src=/workspace/dist,target=/vllm-workspace/dist
 RUN --mount=type=bind,from=flash-attn-builder,src=/usr/src/flash-attention-v2,target=/usr/src/flash-attention-v2 \
     --mount=type=cache,target=/root/.cache/pip \
     pip install /usr/src/flash-attention-v2/*.whl --no-cache-dir
+
+# Manually install flashinfer
+RUN pip install flashinfer -i https://flashinfer.ai/whl/cu121/torch2.3/
```

Comment:
Is this test-specific docker file? or provided to users?

Reply:
We currently cannot include flashinfer to vllm's requirements.txt because it doesn't support extra index URL (https://flashinfer.ai/whl/cu121/torch2.3/). Before FlashInfer is available at PyPI, we can only let users install by themselves.